### PR TITLE
Use `p2p` protocol name over `ipfs` for multiaddr

### LINKED ISSUE #431: Transition /ipfs multiaddr handling to /p2p
See https://github.com/multiformats/multiaddr/pull/64 and https://github.com/multiformats/go-multiaddr/pull/66. 

Places where we manually add `/ipfs/`:
api/util.go:45
api/util.go:61
pstoremgr/pstoremgr.go:106
api/rest/client/client_test.go:63
peer_manager_test.go:36
consensus/raft/consensus_test.go:31
ipfscluster_test.go:238

### PR DESCRIPTION
**Pull request by kishansagathiya**
Fixes #431

### CODE REVIEW

File: api/util.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ func StringsToPeers(strs []string) []peer.ID {
 // peer ID is bad.
```
Comment (hsanjuan) [inline api/util.go]: two lines above, the comment

**Comment by hsanjuan** [comment]
Also, worth checking if p2p addresses are stil being printed as `/ipfs/`, and thus we may want to enable https://godoc.org/github.com/multiformats/go-multiaddr#SwapToP2pMultiaddrs

**Comment by kishansagathiya** [comment]
IPFS addresses haven't been transfered yet
```
➜  ipfs-cluster git:(fix/ipfs-to-p2p) ✗ curl http://localhost:5001/api/v0/id | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   794    0   794    0     0   193k      0 --:--:-- --:--:-- --:--:--  193k
{
  "ID": "QmahdHS1z62tru8iG28Wxw7eZXBzvR5v6QGaumeqKrHuYi",
  "PublicKey": "CAASpgIwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQCoqqnNuL29VSgHNC4T/70qDKucnJNc8X9PudyjnaaV59n74acDfvRXCJySzZEIrSfCXImOY+agMO47/qq57LnQv//MiWWTjzPbkzoY3vGKnDbAkPjsMpSOBaFo+558YJ/P4OP0Ewtu3rH6gtcLLmKD2xL44Ocbl8vvVGAi5vzcCueum1okhvbQr1HJNDBC0nN7C1vHklp1cBPhxS4NXDzOFngPcbBfe/Zsm1Ksno6yyY1lvOyPOeJSBG1hEdQKsQ7t+IhVgBeFcEFtGeTto3Wtkq/hZsgIjCTSVbWs1NWNZTBMZf8xlwc7ynE3btTyL6pev8+GgT7VMELK985Rva0vAgMBAAE=",
  "Addresses": [
    "/ip4/127.0.0.1/tcp/4001/ipfs/QmahdHS1z62tru8iG28Wxw7eZXBzvR5v6QGaumeqKrHuYi",
    "/ip4/172.17.0.2/tcp/4001/ipfs/QmahdHS1z62tru8iG28Wxw7eZXBzvR5v6QGaumeqKrHuYi",
    "/ip4/171.61.93.94/tcp/4001/ipfs/QmahdHS1z62tru8iG28Wxw7eZXBzvR5v6QGaumeqKrHuYi"
  ],
  "AgentVersion": "go-ipfs/0.4.18/6f19dd5",
  "ProtocolVersion": "ipfs/0.1.0"
}
➜  ipfs-cluster git:(fix/ipfs-to-p2p) ✗ 
```

**Comment by kishansagathiya** [comment]
Multiaddr.String() returns a string with `/ipfs` instead of `/p2p`, will need a PR there.

**Comment by hsanjuan** [comment]
@kishansagathiya please use: https://godoc.org/github.com/multiformats/go-multiaddr#SwapToP2pMultiaddrs

**Comment by kishansagathiya** [comment]
It is being used
https://github.com/ipfs/ipfs-cluster/pull/877/files#diff-338f694c7a137597ecc438a5e90b2d89R55
https://github.com/ipfs/ipfs-cluster/pull/877/files#diff-57e0525402652972ffced26d8144aceaR91

**Comment by hsanjuan** [comment]
I misread.

Querying `localhost:5001` shows of course `/ipfs/` multiaddresses as that comes from IPFS. But anything that comes out from cluster comes correctly as `/p2p/`. I am pushing a small change to call the Swap function during `init()` as suggested by the documentation in that function.

**Comment by hsanjuan** [comment]
Last commit moves SwapToP2pMultiaddrs() to the `api` package. It is a better suited place because:

1) It is a single place to do this
2) We are changing essentially how a type is represented in our apis, so it makes sense to do it in the place we handle types.

**Pull request merged**